### ISSUE TITLE
Does not property listen to log events in component runner

### ISSUE THREAD

**Issue by davidturissini** [nils-hoyer/cypress-fail-on-console-error#66]
Inside of https://github.com/nils-hoyer/cypress-fail-on-console-error/blob/main/src/index.ts#L16, `cypress-fail-on-console-error` is listening to `window:before:load` events, which do not get called in component mode. 

Repro:
Add the following to support.ts in a component runner:
```ts
import failOnConsoleError rom "cypress-fail-on-console-error"

failOnConsoleError();
```
Create this test:

```ts
it('should fail', () => {
  console.error('fail');
})
```

Expected: test fail
Actual: test succeeds

**Comment by nils-hoyer** [comment]
Hi @davidturissini, thanks for reporting that issue. Afaik the cypress component runner was in Alpha state in cypress version < 10 and just hit Beta state starting from version 10. I would like to wait until a release version exists before diving into that feature request but feel free to provide a PR or fork the repo.

**Comment by klofi** [comment]
Hello @nils-hoyer, appreciate your work on this plugin. I just tried to use it for a component test that was passing even when there was a console errror, but found the plugin not working and then found this issue.

Cypress component testing is out of beta https://www.cypress.io/blog/2022/11/08/announcing-cypress-11/, any chance you could look into this issue?

**Comment by jimmux** [comment]
I haven't tried using this plugin for component testing yet but while investigating the possibility I noticed that it's setting up the spies on 'window:before:load'. According to Cypress docs, this event is [only used in e2e testing](https://docs.cypress.io/api/cypress-api/catalog-of-events). Looks like a likely cause of the issue.

**Comment by nils-hoyer** [comment]
Linked PR: https://github.com/nils-hoyer/cypress-fail-on-console-error/pull/219

**Comment by nils-hoyer** [comment]
The [PR](https://github.com/nils-hoyer/cypress-fail-on-console-error/pull/219) is merged and support for component testing mode is released with version [5.0.0](https://www.npmjs.com/package/cypress-fail-on-console-error/v/5.0.0). 

**Thanks to @jimmux!**

**Comment by klofi** [comment]
Thank you very much, I can confirm the cypress component tests now fail on console error with v5.0.0.
